File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/client/SpectateFramesPacket.java
package pe.nanamochi.banchus.packets.client;

import java.io.IOException;
import java.io.InputStream;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.components.ReplayFrameBundle;
import pe.nanamochi.banchus.core.ClientPacket;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.io.IDataReader;

@Data
@NoArgsConstructor
public class SpectateFramesPacket implements ClientPacket {
  private ReplayFrameBundle replayFrameBundle;

  @Override
  public Packets getPacketType() {
    return Packets.OSU_SPECTATE_FRAMES;
  }

  @Override
  public void read(IDataReader reader, InputStream stream) throws IOException {
    this.replayFrameBundle = ReplayFrameBundle.read(reader, stream);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/client/SpectateFramesPacket.kt
package pe.nanamochi.banchus.packets.client

import java.io.InputStream
import pe.nanamochi.banchus.components.ReplayFrameBundle
import pe.nanamochi.banchus.core.ClientPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataReader

class SpectateFramesPacket : ClientPacket() {
    override val type = PacketType.OSU_SPECTATE_FRAMES

    var replayFrameBundle: ReplayFrameBundle = ReplayFrameBundle()

    override fun read(reader: DataReader, ins: InputStream) {
        this.replayFrameBundle = ReplayFrameBundle.read(reader, ins)
    }
}
